genePred/compare.go

```diff
@@ -1,61 +1,65 @@
 package genePred
 
 import (
-	"log"
 	"strings"
 )
 
-func AllAreEqual(a []*GenePred, b []*GenePred) bool {
+func AllAreEqual(a []GenePred, b []GenePred) (bool, []int) {
```

Comment:
Returning the []int to give the programmer more detailed information on what didn't match is an interesting idea, but because it is non-standard, I think this function and Equal need to have comments that explain to the user what they are getting back.  It is pretty clear what they get from the code, but the goal is for the documentation to be able to stand on its own without having the person need to look at the actual code.
Is this more for debugging so that the human can know what didn't match (but was anticipated to match)?  Or is the idea that you have a program that needs to know what didn't match and then will do something different if the name didn't match or if the start position didn't match?  If the first case, maybe it makes sense to put this in a _test.go file with error messages that describe which comparison failed?

Reply:
my idea is the second, where in a real-world scenario you'd want to know if two entries are different.